partiql-spi/src/main/kotlin/org/partiql/spi/connector/sql/exts/FnExists.kt

```diff
@@ -0,0 +1,107 @@
+package org.partiql.spi.connector.sql.exts
+
+import org.partiql.spi.fn.Fn
+import org.partiql.spi.fn.FnExperimental
+import org.partiql.spi.fn.FnParameter
+import org.partiql.spi.fn.FnSignature
+import org.partiql.value.BagValue
+import org.partiql.value.ListValue
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.SexpValue
+import org.partiql.value.StructValue
+import org.partiql.value.boolValue
+import org.partiql.value.check
+
+@OptIn(PartiQLValueExperimental::class, FnExperimental::class)
+internal object Fn_EXISTS__BAG__BOOL : Fn {
+
+    override val signature = FnSignature(
+        name = "exists",
+        returns = PartiQLValueType.BOOL,
+        parameters = listOf(
+            FnParameter("container", PartiQLValueType.BAG),
+        ),
+        isNullCall = true,
+        isNullable = false,
+    )
+
+    override fun invoke(args: Array<PartiQLValue>): PartiQLValue {
+        val container = args[0].check<BagValue<*>>()
+        val iter = container.iterator()
+        if (iter.hasNext()) {
+            return boolValue(true)
+        }
+        return boolValue(false)
```

Comment:
```suggestion
        val exists = container.iterator().hasNext()
        return boolValue(exists)
```

Nit -- non-blocking, but just a recommendation